extensions/wikia/AdEngine/js/video/videoFrequencyMonitor.js

```diff
@@ -0,0 +1,39 @@
+/*global define*/
+define('ext.wikia.adEngine.video.videoFrequencyMonitor', [
+	'ext.wikia.adEngine.adLogicPageViewCounter',
+	'ext.wikia.adEngine.video.videoFrequencySettings',
+	'ext.wikia.adEngine.video.videoFrequencyStore'
+], function (pageViewCounter, settings, store) {
+	'use strict';
+
+	function prepareData() {
+		return {
+			date: new Date().getTime(),
+			pv: pageViewCounter.get()
+		};
+	}
+
+	function registerLaunchedVideo() {
+		store.save(prepareData());
+	}
+
+	function videoCanBeLaunched() {
+		var limits = settings.get(),
+			result = true;
+
+		limits.pv.map(function (rule) {
```

Comment:
`map`? Are we collecting result anywhere? Maybe it's better to use `forEach` here?